packages/shared/src/provider/provider.ts

```diff
@@ -82,7 +82,7 @@ export { ClientProviderStatus as AllProviderStatus };
 /**
  * Static data about the provider.
  */
-export interface ProviderMetadata extends Metadata {
+export interface ProviderMetadata extends Readonly<Metadata> {
```

Comment:
Will read-only prevent providers from updating metadata? The metadata should be read-only to consumers but I think a provider should be able to update metadata at runtime.

Reply:
I'll double check to see if TS prevents it for consumers and allows it for the provider. I am almost certain the provider can still set its own data. I had read the comment for `ProviderMetadata` which says it's static and so kept it `Readonly`.

Reply:
This won't cause an issue for providers except if they want to write/mutate inner props of metadata, but this can be easily worked-around.